src/components/PeriodDimension/utils/fixedPeriods.js

```diff
@@ -40,510 +41,182 @@ const PERIOD_TYPE_REGEX = {
+            config,
+            fnFilter,
+            calendar,
+        })
     }
 }
 
-const getYearlyPeriodType = (formatYyyyMmDd, fnFilter) => {
+const getSixMonthlyAprilPeriodType = (fnFilter, calendar) => {
     return (config) => {
-        let periods = []
-        const offset = parseInt(config.offset, 10)
-        const isFilter = config.filterFuturePeriods
-        const isReverse = config.reversePeriods
-        const year = new Date(Date.now()).getFullYear() + offset
-        const date = new Date(`31 Dec ${year}`)
-
-        while (year - date.getFullYear() < 10) {
-            const period = {}
-            period.endDate = formatYyyyMmDd(date)
-            date.setMonth(0, 1)
-            period.startDate = formatYyyyMmDd(date)
-            const dateString = date.getFullYear().toString()
-            period.name = dateString
-            period.iso = date.getFullYear().toString()
-            period.id = period.iso.toString()
-            periods.push(period)
-            date.setDate(0)
-        }
-
-        periods = isFilter ? fnFilter(periods) : periods
-        periods = isReverse ? periods : periods.reverse()
-        // Years are collected backwards. If isReverse is true, then do nothing. Else reverse to correct order and return.
-
-        return periods
+        return getPeriods({
+            periodType: 'SIXMONTHLYAPR',
+            config,
+            fnFilter,
+            calendar,
+        })
     }
 }
 
-const getFinancialOctoberPeriodType = (formatYyyyMmDd, fnFilter) => {
+const getYearlyPeriodType = (fnFilter, calendar) => {
     return (config) => {
-        let periods = []
-        const offset = parseInt(config.offset, 10)
-        const isFilter = config.filterFuturePeriods
-        const isReverse = config.reversePeriods
-        const year = new Date(Date.now()).getFullYear() + offset
-        const date = new Date(`30 Sep ${year + 1}`)
-
-        for (let i = 0; i < 10; i++) {
-            const period = {}
-            period.endDate = formatYyyyMmDd(date)
-            date.setYear(date.getFullYear() - 1)
-            date.setDate(date.getDate() + 1)
-            period.startDate = formatYyyyMmDd(date)
-            const yearStart = date.getFullYear()
-            const yearEnd = date.getFullYear() + 1
-            period.name = `${getMonthName(9)} ${yearStart} - ${getMonthName(
-                8
-            )} ${yearEnd}`
-            period.id = `${date.getFullYear()}Oct`
-            periods.push(period)
-            date.setDate(date.getDate() - 1)
-        }
-
-        periods = isFilter ? fnFilter(periods) : periods
-        periods = isReverse ? periods : periods.reverse()
-        // FinancialOctober periods are collected backwards. If isReverse is true, then do nothing. Else reverse to correct order and return.
-
-        return periods
+        return getPeriods({
+            periodType: 'YEARLY',
+            config: {
+                ...config,
+                reversePeriods: true,
```

Comment:
Why is this overridden?

Reply:
Ok, I wasn't sure because in the old code `reversePeriod` was passed in the config, and the config is used also in the new code.